src/apps/allbridge/positions.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  PositionsHook,
+  AppTokenPositionDefinition,
+  UnknownAppTokenError,
+  TokenDefinition,
+  ContractPositionDefinition,
+} from '../../types/positions'
+import { Address } from 'viem'
+import { DecimalNumber, toDecimalNumber } from '../../types/numbers'
+import BigNumber from 'bignumber.js'
+import { getClient } from '../../runtime/client'
+import { getTokenId } from '../../runtime/getTokenId'
+import { getAllBridgeTokenInfo } from './api'
+import { poolAbi } from './abis/pool'
+
+const ALLBRIDGE_LOGO =
+  'https://raw.githubusercontent.com/valora-inc/dapp-list/main/assets/allbridgecore.png'
+
+const hook: PositionsHook = {
+  getInfo() {
+    return {
+      id: 'allbridge',
+      name: 'Allbridge',
+      description: '',
+    }
+  },
+  async getPositionDefinitions(networkId, address) {
+    const allbridgeTokenInfo = (await getAllBridgeTokenInfo({ networkId }))
+      .tokens
+
+    const client = getClient(networkId)
+
+    const balances = await Promise.all(
+      allbridgeTokenInfo.map(async (tokenInfo) => {
+        return address
+          ? client.readContract({
+              address: tokenInfo.poolAddress,
+              abi: poolAbi,
+              functionName: 'balanceOf',
+              args: [address as Address],
+            })
+          : undefined
+      }),
+    )
+
+    const rewards = await Promise.all(
+      allbridgeTokenInfo.map(async (tokenInfo) => {
+        return address
+          ? client.readContract({
+              address: tokenInfo.poolAddress,
+              abi: poolAbi,
+              functionName: 'pendingReward',
+              args: [address as Address],
+            })
+          : undefined
+      }),
+    )
+
+    return allbridgeTokenInfo.flatMap((tokenInfo, i) => {
+      const apr = new BigNumber(tokenInfo.apr7d).toNumber() * 100
+
+      const balanceOf = balances[i]
+      const pendingReward = rewards[i]
+
+      const useAToken = !balanceOf || balanceOf > 0
+      const showReward = !!pendingReward && pendingReward > 0
+      return [
+        useAToken &&
+          ({
+            type: 'app-token-definition',
+            networkId,
+            address: tokenInfo.poolAddress.toLowerCase(),
+            tokens: [
+              { address: tokenInfo.tokenAddress.toLowerCase(), networkId },
+            ],
+            availableShortcutIds: ['deposit', 'withdraw'],
+            displayProps: {
+              title: tokenInfo.symbol,
+              description: `Supplied (APR: ${apr.toFixed(2)}%)`,
+              imageUrl: ALLBRIDGE_LOGO,
+            },
+            dataProps: {
+              apy: apr,
```

Comment:
should we make a different field for apr to make it explicit? Related to https://github.com/valora-inc/wallet/pull/5679#discussion_r1692049812

Reply:
added an apr field to dataProps and made both optional